src/sources/MVT.js

```diff
@@ -157,7 +157,7 @@ export default class MVT extends Base {
         const mvtLayer = tile.layers[this._options.layerID || Object.keys(tile.layers)[0]];
 
         if (!mvtLayer) {
-            throw new Error(`LayerID '${this._options.layerID}' doesn't exist on the MVT tile`);
+            return { empty: true };
```

Comment:
I don't know if this is enough.

If you have `layer0` and `layer1` as layer IDs, the tile has data for `layer0` but not for `layer1`, you ask for data for `layer1`, then `tile.layers[this._options.layerID || Object.keys(tile.layers)[0]]` will put the data from `layer0` in `mvtLayer`, it won't be a false-ish value but it will contain data for the incorrect layer.

Reply:
Yes but in that case, the user is doing something that is explicitly documented as forbidden. The first tile with both layers will throw in that example.

Also, is there a way to control that without removing the functionality of allowing users to not define a layerID when there is only one layer? The way I see it, this is a limitation of the MVT format which doesn't include the layer name of layers that have no features. Am I missing somthing?

Reply:
The user doesn't know in advance if different tiles will or will not have data for one of the layers.

As an example, you could have a two layers tileset, accessible through `example.com/{z}/{x}/{y}.mvt`, where different tiles are like:

```
tile z=0,x=0,y=0:
  -layer0: data-data-data
  -layer1: data-data

tile z=1,x=0,y=0:
  -layer1: data-data-data-data

tile z=1,x=1,y=1:
  -layer0: data
```

My point is:
 - If the user specified a `layerID`, we cannot assume there is no layer like that, in that case, we need to return `{ empty: true }` as the only alternative because we might be in one of the previous scenarios.
 - If no `layerID` is provided, we will attempt to access the first layer, if there is no data: first, we should not reach this part of the code anyway; in the case we do, we will again return an `{ empty: true }`. 🤔
 - But with the current code we could be getting data from another layer (the first layer found in the tile) if there is no data for the requested `layerID`.

Does that make sense?

Reply:
> But with the current code we could be getting data from another layer (the first layer found in the tile) if there is no data for the requested layerID.

Yes, with the current code that could happen, but I think it is inevitable unless we remove the ability to use one single layer without specifying its ID, no?

Reply:
My proposal is just a bit different:

```diff
diff --git a/src/sources/MVT.js b/src/sources/MVT.js
index 5657fb6e..b8abdbc4 100644
--- a/src/sources/MVT.js
+++ b/src/sources/MVT.js
@@ -154,10 +154,10 @@ export default class MVT extends Base {
             throw new Error(`LayerID parameter wasn't specified and the MVT tile contains multiple layers: ${JSON.stringify(Object.keys(tile.layers))}`);
         }

-        const mvtLayer = tile.layers[this._options.layerID || Object.keys(tile.layers)[0]];
+        const mvtLayer = this._options.layerID ? tile.layers[this._options.layerID] : Object.keys(tile.layers)[0];

         if (!mvtLayer) {
-            throw new Error(`LayerID '${this._options.layerID}' doesn't exist on the MVT tile`);
+            return { empty: true };
         }

         const { geometries, properties, numFeatures } = this._decodeMVTLayer(mvtLayer, this._metadata, MVT_EXTENT);
```

Reply:
I still don't understand it. The second change is the same as the one I did, the first one though... I don't understand its intent, and I think it's broken too: `Object.keys(tile.layers)[0]` will return the layer name I think, but you need the layer object. Changing `keys` by `values` will solve that, but then, I think the code behaves the same as the original one.

Reply:
Sorry, I obviously wrote a bug, I meant:

```js
const mvtLayer = this._options.layerID ? tile.layers[this._options.layerID] : tile.layers[Object.keys(tile.layers)[0]];
```

Reply:
So, if there is a layerID, you retrieve by IayerID, if there is no layerID you use the first layer. Previously there is a check for multiple layers. That way we avoid getting data from another layer if there is a layerID specified.

Reply:
But previously you have `const mvtLayer = tile.layers[this._options.layerID || Object.keys(tile.layers)[0]];` which I think it's the same right?

Reply:
I was thinking all the time about `layerID` being undefined.  🤦‍♂️ 

You are right.

Now I get your point in:
> Yes, with the current code that could happen, but I think it is inevitable unless we remove the ability to use one single layer without specifying its ID, no?

Sorry for the noise.